aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -855,6 +894,9 @@ public AwsProperties(Map<String, String> properties) {
 
     this.s3SignerImpl = properties.get(S3_SIGNER_IMPL);
     this.allProperties = SerializableMap.copyOf(properties);
+    this.credentialsProviderProperties =
+        PropertyUtil.filterProperties(
+            properties, key -> key.startsWith(CREDENTIAL_PROVIDER_PREFIX));
```

Comment:
I think we may want to use `PropertyUtil.propertiesWithPrefix` instead of `PropertyUtil.filterProperties`

`PropertyUtil.propertiesWithPrefix` trims out the prefix of the keys:
```java
client.credentials-provider.my_key=my_value
```
becomes
```java
my_key: my_value
```

I assume that when passing properties into the `AwsCredentialsProvider.create(Map)`, the prefix `client.credentials-provider` is not wanted. But since I did not find any existing implementation under https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/auth/credentials/AwsCredentialsProvider.html that contains the method `create(Map)`, I may be wrong. Just want to make sure the properties are correctly set.